plugin/src/main/scala-3/chisel3/internal/plugin/BundlePhase.scala

```diff
@@ -0,0 +1,227 @@
+    val bundleSym = record.symbol.asClass
+    val recordTpe = requiredClass("chisel3.Record")
+    def isBundleDataField(m: Symbol): Boolean = {
+      m.isPublic
+      && (
+        ChiselTypeHelpers.isData(m.info)
+          || ChiselTypeHelpers.isBoxedData(m.info)
+      )
+    }
+
+    val currentFields: List[Tree] = bundleSym.info.decls.toList.collect {
+      case m if isBundleDataField(m) =>
+        val name = m.name.show.trim
+        val thisRef: tpd.Tree = tpd.This(bundleSym.asClass)
+        val sel:     tpd.Tree = tpd.Select(thisRef, m.termRef)
+        tupleTree(List(tpd.Literal(Constant(name)), sel))
+    }
+    currentFields
+  }
+
+  def generateElements(record: tpd.TypeDef)(using Context): tpd.DefDef = {
+    val bundleSym = record.symbol.asClass
+
+    val currentFields: List[Tree] = getBundleFields(record)
+
+    val dataTpe = requiredClassRef("chisel3.Data")
+    val rhs = makeArray(currentFields)
+
+    val tupleTpe =
+      defn.TupleClass.typeRef.appliedTo(List(defn.StringType, defn.AnyType))
+
+    val iterableTpe =
+      requiredClassRef("scala.collection.Iterable").appliedTo(tupleTpe)
+
+    val elementsSym: Symbol =
+      newSymbol(
+        bundleSym,
+        Names.termName("_elementsImpl"),
+        Flags.Method | Flags.Override | Flags.Protected,
+        MethodType(Nil, Nil, iterableTpe)
+      )
+
+    tpd.DefDef(elementsSym.asTerm, rhs)
+  }
+}
+
+class ChiselBundlePhase extends PluginPhase {
+  val phaseName: String = "chiselBundlePhase"
+  override val runsAfter = Set(TyperPhase.name)
+
+  override def transformTypeDef(record: tpd.TypeDef)(using Context): tpd.Tree = {
+    if (
+      ChiselTypeHelpers.isRecord(record.tpe)
+      && !record.symbol.flags.is(Flags.Abstract)
+    ) {
+      val isBundle: Boolean = ChiselTypeHelpers.isBundle(record.tpe)
+      val thiz:     tpd.This = tpd.This(record.symbol.asClass)
+
+      // // ==================== Generate _cloneTypeImpl ====================
+      val conArgs: Option[List[List[tpd.Tree]]] =
+        BundleHelpers.extractConArgs(record, thiz, isBundle)
+      val cloneTypeImplOpt =
+        BundleHelpers.generateAutoCloneType(record, thiz, conArgs, isBundle)
+
+      // ==================== Generate val elements (Bundles only) ====================
+      val elementsImplOpt: Option[tpd.DefDef] =
+        if (isBundle) Some(BundleHelpers.generateElements(record)) else None
+
+      // ==================== Generate _usingPlugin ====================
+      val usingPluginOpt =
+        if (isBundle) {
+          val isPluginSym = newSymbol(
+            record.symbol,
+            Names.termName("_usingPlugin"),
+            Flags.Method | Flags.Override | Flags.Protected,
+            defn.BooleanType
+          )
+          Some(
+            tpd.DefDef(
+              isPluginSym.asTerm,
+              _ => tpd.Literal(Constant(true))
+            )
+          )
+        } else None
```

Comment:
```suggestion
        Option.when(isBundle) {
          val isPluginSym = newSymbol(
            record.symbol,
            Names.termName("_usingPlugin"),
            Flags.Method | Flags.Override | Flags.Protected,
            defn.BooleanType
          )
          Some(
            tpd.DefDef(
              isPluginSym.asTerm,
              _ => tpd.Literal(Constant(true))
            )
          )
        }
```
Style nit but `Option.when` is great, please use it especially when multiline.